app/src/main/java/com/example/henzoshimada/feeltrip/ElasticSearchController.java

```diff
@@ -527,11 +537,11 @@ public GetFilteredMoodsTask(String searchmode){ // must pass this specific set o
             }
 
             if(emotionfilter) {
-                query += "\"must\" : { \"term\" : { \"emotionalState\" : \"" + emotion + "\" }},";
+                query += "\"must\" : { \"match\" : { \"emotionalState\" : \"" + emotion + "\" }},";
             }
 
             if(keywordfilter) {
-                query += "\"must\" : { \"term\" : { \"description\" : \"" + search_parameters[0] + "\" }},";
+                query += "\"must\" : { \"match\" : { \"description\" : \"" + keyword + "\" }},";
```

Comment:
Turns out this query accepts more than just one keyword, simply separate the keyterms with spaces. It doesn't search for keyphrases though, which is fine because we desire keyword searches only anyways.